app/controllers/users/sessions_controller.rb

```diff
@@ -79,6 +80,40 @@ def process_locked_out_session
       redirect_to root_url
     end
 
+    def valid_captcha_result?
+      if cookies[:device] && (device = Device.find_by(cookie_uuid: cookies[:device]))
```

Comment:
Discussed a little bit offline, but this query should probably include a subquery on email address for the user_id since `cookie_uuid` is not unique.